packages/ember-toucan-core/src/components/form/controls/multiselect.ts

```diff
@@ -176,6 +198,50 @@ export default class ToucanFormMultiselectControlComponent extends Component<Tou
     return `${this.popoverId}-${this.activeIndex}`;
   }
 
+  /**
+   * Convenience getter for determining if our "Select all" feature
+   * is enabled.
+   */
+  get isSelectAllEnabled() {
+    return Boolean(this.args.selectAllText);
+  }
+
+  /**
+   * Helps us determine if the "Select all" option should be
+   * visible in the UI or not based on our rules above in the component
+   * argument section.
+   */
+  get isSelectAllVisible() {
+    return this.isSelectAllEnabled && this.inputValue?.length === 0;
+  }
+
+  /**
+   * Helps us determine if the "Select all" option checkbox
+   * should be checked or not.
+   */
+  get isSelectAllChecked() {
+    // We need to use this comparison method when comparing arrays
+    // as Ember's `isEqual` is only for objects.
+    return (
+      JSON.stringify(this.selected?.sort()) ===
+      JSON.stringify(this.options?.sort())
+    );
```

Comment:
Open to some code golf here, but we can't use [Ember's `isEqual`](https://api.emberjs.com/ember/release/functions/@ember%2Futils/isEqual) because it doesn't play well with comparing arrays.

> Due to the expense of array comparisons, collections will never be equal to each other even if each of their items are equal to each other.

Sorting `selected` and `options` and then doing a JSON stringify comparison is the quickest way I know to do this.  Since we know there won't be duplicates in the arrays (since our business logic doesn't allow for that - re-selecting an already selected item removes it) I think this is probably safe, but I'm also open to other suggestions if anyone feels strongly about improving this.

Reply:
- [ ] Add this comment in code!

Reply:
Seems like the simplest way. IMO not worth bringing in Lodash just for this.